addons/sale/models/sale_order_line.py

```diff
@@ -595,43 +611,47 @@ def _get_display_price(self, product):
         # negative discounts (= surcharge) are included in the display price
         return max(base_price, final_price)
 
+    @api.depends('product_id')
+    def _compute_name(self):
+        for line in self:
+            if not line.product_id:
+                continue
+            product = line.product_id.with_context(
+                lang=get_lang(line.env, line.order_id.partner_id.lang).code,
+                partner=line.order_id.partner_id,
+                quantity=line.product_uom_qty,
+                date=line.order_id.date_order,
+                pricelist=line.order_id.pricelist_id.id,
+                uom=line.product_uom.id)
```

Comment:
```suggestion
            product = line.product_id.with_context(
                lang=get_lang(line.env, line.order_id.partner_id.lang).code)
```

Only the lang context is necessary for the description re-computation :)